### PR TITLE
Migrate launch-chrome.sh to use chrome-launcher.

### PR DESCRIPTION
**Pull request by samccone**

### CODE REVIEW

**Comment by paulirish** [comment]
image

**Comment by samccone** [comment]
refetch

**Comment by paulirish** [comment]
k thx

File: lighthouse-cli/manual-chrome-launcher.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+require('./compiled-check.js')('chrome-launcher.js');
+
+const args = process.argv.slice(2);
+
+let chromeArgs;
+let startingUrl;
+
+if (args.length) {
+  // If the first argument starts with -- assume all are chrome args.
+  if (args[0].startsWith('--')) {
```
Comment (paulirish) [inline lighthouse-cli/manual-chrome-launcher.js]: This assumption is a little restrictive IMO.. Launching chrome as `chrome --flags http://url` is the most typical way, so I think we should preserve that.

Let's send everything that `startsWith('-')` as `additionalFlags` and send any remaining args as the startingURL string. 
(Technically multiple URLs will open multiple tabs and that should work already via chromeLauncher)

File: lighthouse-cli/chrome-launcher.ts [hunk 2]
```diff
@@ -39,14 +39,22 @@ class ChromeLauncher {
...
   pidFile: string
+  startingUrl: string
+  additionalFlags: Array<string>
   chrome: childProcess.ChildProcess
 
   // We can not use default args here due to support node pre 6.
-  constructor(opts?: {autoSelectChrome?: Boolean}) {
-    opts = opts || {};
+  constructor(opts?: {
+      url?: string,
```
Comment (paulirish) [inline lighthouse-cli/chrome-launcher.ts]: wanna rename `url` to `startingUrl` so we have consistency?

File: lighthouse-cli/manual-chrome-launcher.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env node
+
+'use strict'
```
Comment (paulirish) [inline lighthouse-cli/manual-chrome-launcher.js]: let's add a little comment for usage details:

```sh
/**
 * @fileoverview Script to launch a clean Chrome instance on-demand.
 * Assuming Lighthouse is installed globally or `npm link`ed, use via:
 *     chrome-debug
 * Optionally pass additional flags and/or a URL
 *     chrome-debug http://goat.com
 *     chrome-debug --show-paint-rects
 */
```

File: lighthouse-cli/manual-chrome-launcher.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  // If the first argument starts with -- assume all are chrome args.
+  if (args[0].startsWith('--')) {
+    chromeArgs = args;
+  } else {
+    startingUrl = args[0];
+    chromeArgs = args.slice(1);
+  }
+}
+
+const chromeLauncher = require('./chrome-launcher.js').ChromeLauncher;
```
Comment (paulirish) [inline lighthouse-cli/manual-chrome-launcher.js]: Capital `ChromeLauncher` as its a class not an instance, yah?

File: lighthouse-cli/manual-chrome-launcher.js (hunk 3 above)
Comment (samccone) [inline lighthouse-cli/manual-chrome-launcher.js]: done

File: lighthouse-cli/chrome-launcher.ts (hunk 2 above)
Comment (samccone) [inline lighthouse-cli/chrome-launcher.ts]: done

File: lighthouse-cli/manual-chrome-launcher.js (hunk 1 above)
Comment (samccone) [inline lighthouse-cli/manual-chrome-launcher.js]: > (Technically multiple URLs will open multiple tabs and that should work already via chromeLauncher)

This does not already work

Comment (paulirish) [inline lighthouse-cli/manual-chrome-launcher.js]: I meant it should work already with launch-chrome.sh, rather. whoops.

i dont feel strongly that we need to support opening multiple via this, so w/e.
what you did in the latest commit sgtm

**Pull request merged**